python/foxglove-sdk/python/docs/api/index.rst

```diff
@@ -3,12 +3,179 @@ API Reference
+
+.. py:function:: create_notebook_buffer(context: Context | None = None) -> None
+
+   Create a data buffer for collecting messages in Jupyter notebooks. The buffer
+   will be associated with the provided context, so every message logged to the context
+   will be collected by the buffer.
+
+   :param context: Optional Foxglove context to use for logging. If not provided,
+       the global context will be used. This allows you to isolate the
+       buffer's data collection from other parts of your application.
+
+   .. note::
+      This function is only available when the `notebook` extra package
+      is installed.
+
+.. py:function:: visualize(context: Optional[Context] = None, width: Optional[str] = None, height: Optional[str] = None, src: Optional[str] = None, layout_data: Optional[dict] = None) -> FoxgloveViewer
+
+   Create a FoxgloveViewer widget for interactive data visualization in Jupyter notebooks.
+
+   This function creates an embedded Foxglove visualization widget that displays
+   the data collected in the NotebookBuffer associated with the provided context.
+   The widget provides a fully-featured Foxglove interface directly within
+   your Jupyter notebook, allowing you to explore multi-modal robotics data
+   including 3D scenes, plots, images, and more.
+
+   :param context: The Context used to log the messages. If no Context is provided, the global
+       context will be used. The visualization data will be retrieved from the NotebookBuffer
+       associated with the provided context. This buffer should have been populated with
+       logged messages before creating the viewer.
+   :param width: Optional width for the widget. Can be specified as CSS values like
+       "800px", "100%", "50vw", etc. If not provided, defaults to "100%".
+   :param height: Optional height for the widget. Can be specified as CSS values like
+       "600px", "80vh", "400px", etc. If not provided, defaults to "500px".
+   :param src: Optional URL of the Foxglove app instance to use. If not provided or empty,
+       uses the default Foxglove embed server (https://embed.foxglove.dev/).
+   :param layout_data: Optional layout data to be used by the Foxglove viewer. Should be a
+       dictionary that was exported from the Foxglove app. If not provided, uses the
+       default layout.
+   :return: A Jupyter widget that displays the embedded Foxglove
+       visualization interface with the provided data.
+   :raises Exception: If the notebook extra package is not installed. Install it
+       with `pip install foxglove-sdk[notebook]`.
+
+   .. note::
+      This function is only available when the `notebook` extra package
+      is installed. Install it with `pip install foxglove-sdk[notebook]`.
+      The widget will automatically load the data from the provided datasource
+      and display it in the embedded Foxglove viewer.
+
+Notebook Classes
+^^^^^^^^^^^^^^^^
+
+.. py:class:: NotebookBuffer
```

Comment:
I'm not convinced we should put this in the documentation. Is it really part of the public API?